src/windows.rs

```diff
@@ -127,17 +133,17 @@ impl Client {
         ))
     }
 
-    pub unsafe fn open(s: &str) -> Option<Client> {
+    pub unsafe fn open(s: &str) -> Result<Client, ErrFromEnv> {
         let name = match CString::new(s) {
             Ok(s) => s,
-            Err(_) => return None,
+            Err(_) => return Err(ErrFromEnv::IsNotCString),
         };
 
         let sem = OpenSemaphoreA(SYNCHRONIZE | SEMAPHORE_MODIFY_STATE, FALSE, name.as_ptr());
         if sem.is_null() {
-            None
+            Err(ErrFromEnv::CannotAcquireSemaphore)
```

Comment:
Wouldn't it be better to return `io::Error` here, which will give much better error message?